test/rpc/test_data/get_events.py

```diff
@@ -8,120 +8,65 @@
-    ],
-    "chunk_size": "test",
-}
-
-BLOCK_FROM_0_TO_LATEST_MISSING_PARAMETER = {
-    "from_block": {"block_number": 0},
-    "keys": [
-        rpc_felt(FEE_CHARGED_EVENT_KEY),
-    ],
-    "to_block": "latest",
-    "chunk_size": 10,
-}
-
-BLOCK_FROM_0_TO_LATEST_WRONG_BLOCK_TYPE = {
-    "from_block": {"block_number": "0x0"},
-    "address": rpc_felt(EXPECTED_FEE_TOKEN_ADDRESS),
-    "keys": [
-        rpc_felt(FEE_CHARGED_EVENT_KEY),
-    ],
-    "to_block": "latest",
-    "chunk_size": 10,
-}
-
-BLOCK_FROM_0_TO_2 = {
-    "from_block": {"block_hash": rpc_felt("0x0")},
-    "to_block": {"block_number": 2},
-    "address": rpc_felt(EXPECTED_FEE_TOKEN_ADDRESS),
-    "keys": [
-        rpc_felt(FEE_CHARGED_EVENT_KEY),
-    ],
-    "chunk_size": 10,
-}
+def create_get_events_filter(
+    *,
+    from_block: Union[int, str] = 0,
+    to_block: Union[int, str] = "latest",
+    address: Optional[str] = rpc_felt(EXPECTED_FEE_TOKEN_ADDRESS),
+    keys: Optional[List[str]] = None,
+    chunk_size: int = 10,
+    continuation_token: Optional[str] = None
+) -> dict:
+    """
+    Method that creates ``filter`` parameter for `get_events` RPC call.
+
+    :param from_block: Identifier of the block (number of hash) from which the events start.
+        The value can be a rpc felt, an int or literals "latest" or "pending". Defaults to 0.
+    :param to_block: Identifier of the block (number or hash) to which the events end.
+        The value can be a rpc felt, an int or literals "latest" or "pending". Defaults to "latest".
+    :param address: Address of the contract.
+    :param keys: Names of events that are searched for in rpc felt form.
+    :param chunk_size: Size of returned one chunk of events, defaults to 10.
+    :param continuation_token: (optional) String with a continuation token.
+
+    :return: `filter` param matching the specification.
+    """
+    if keys is None:
+        keys = [rpc_felt(FEE_CHARGED_EVENT_KEY)]
+    filter_body = {
+        "from_block": parse_block_delimiter_parameter(from_block),
+        "keys": keys,
+        "to_block": parse_block_delimiter_parameter(to_block),
+        "chunk_size": chunk_size,
+    }
+    if address is not None:
+        filter_body["address"] = address
+    if continuation_token is not None:
+        filter_body["continuation_token"] = continuation_token
+    return {"filter": filter_body}
 
-BLOCK_FROM_3_TO_3 = {
-    "from_block": {"block_number": 3},
-    "to_block": {"block_number": 3},
-    "address": rpc_felt(EXPECTED_FEE_TOKEN_ADDRESS),
-    "keys": [
-        rpc_felt(FEE_CHARGED_EVENT_KEY),
-    ],
-    "chunk_size": 10,
-}
 
-BLOCK_FROM_2_TO_3 = {
-    "from_block": {"block_number": 2},
-    "to_block": {"block_number": 3},
-    "address": rpc_felt(EXPECTED_FEE_TOKEN_ADDRESS),
-    "keys": [
-        rpc_felt(FEE_CHARGED_EVENT_KEY),
-    ],
-    "chunk_size": 10,
-}
+def parse_block_delimiter_parameter(
+        block_delimiter: Union[str, int]
```

Comment:
There is a block identifier type in starkware.starknet.services.api.feeder_gateway.response_objects, can that be used here?

Reply:
I think it works, for the sake of clarity I'd make that something like `Union[BlockIdentifier, BlockHash]`, where `BlockHash = str`.

Reply:
Right, now I see we can't just use `BlockIdentifier` since we want to support hashes. Regarding `BlockHash`, that type is defined somewhere (rpc/structures/types.py)